operators/src/source/gdal_source.rs

```diff
@@ -372,6 +374,155 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+                    time_steps_no_gaps.push(TimeInterval::new(
+                        (time.end() + minimum_step)?,
+                        sorted_time_steps[index + 1].start(),
+                    )?);
+                } else {
+                    time_steps_no_gaps.push(TimeInterval::new(
+                        time.end(),
+                        sorted_time_steps[index + 1].start(),
+                    )?);
+                }
+            }
+        }
+
+        time_steps_no_gaps.sort_by(|a, b| a.start().partial_cmp(&b.start()).unwrap());
+
+        let valid_duration = TimeInterval::new(
+            time_steps_no_gaps[0].start(),
+            time_steps_no_gaps[time_steps_no_gaps.len() - 1].start(),
+        )?;
+
+        let parts = match valid_duration.intersect(&query.time_interval) {
+            Some(time_interval) => {
+                let mut loading_info_parts: Vec<GdalLoadingInfoPart> = Vec::new();
+
+                if query.time_interval.start() < time_interval.start() {
+                    let gap =
+                        TimeInterval::new(query.time_interval.start(), time_interval.start())?;
+
+                    loading_info_parts.push(
+                        self.params
+                            .replace_time_placeholders(&self.time_placeholders, gap)
+                            .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                time: gap,
+                                params: loading_info_part_params,
+                            })?,
+                    );
+                }
+                for time_position in &time_steps_no_gaps {
+                    if let Some(step_interval) = time_interval.intersect(time_position) {
+                        loading_info_parts.push(
+                            self.params
+                                .replace_time_placeholders(&self.time_placeholders, step_interval)
+                                .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                    time: step_interval,
+                                    params: loading_info_part_params,
+                                })?,
+                        );
+                    }
+                }
+
+                if query.time_interval.end() > time_interval.end() {
+                    let gap = TimeInterval::new(time_interval.end(), query.time_interval.end())?;
+
+                    loading_info_parts.push(
+                        self.params
+                            .replace_time_placeholders(&self.time_placeholders, gap)
+                            .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                time: gap,
+                                params: loading_info_part_params,
+                            })?,
+                    );
+                }
+
+                loading_info_parts
+            }
+            .into_iter(),
+
+            None => vec![self
```

Comment:
if there is no intersection, the source should just create no-data tiles instead of trying to load data.